xarray/core/merge.py

```diff
@@ -564,7 +564,7 @@ def merge_coords(
 
 def assert_valid_explicit_coords(
     variables: Mapping[Any, Any],
-    dims: Mapping[Any, int],
+    dims: Iterable[Any],
```

Comment:
I changed this because the lengths are never used, and renamed `dims`->`sizes` in a few places to avoid this confusion.